src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry2.java

```diff
@@ -0,0 +1,507 @@
+        }
+    }
+
+    private static void deserializeTRSR(JsonDeserializationContext context, EnumMap<ItemCameraTransforms.TransformType, TRSRTransformation> transforms, JsonObject transform, String name, ItemCameraTransforms.TransformType itemCameraTransform)
+    {
+        if(transform.has(name))
+        {
+            TRSRTransformation t = context.deserialize(transform.remove(name), TRSRTransformation.class);
+            transforms.put(itemCameraTransform, TRSRTransformation.blockCenterToCorner(t));
+        }
+    }
+
+    public static IBakedModel bakeHelper(BlockModel blockModel, ModelBakery modelBakery, BlockModel otherModel, Function<ResourceLocation, TextureAtlasSprite> spriteGetter, ISprite sprite, VertexFormat format)
+    {
+        IBakedModel model;
+        IModelGeometry<?> customModel = blockModel.customData.getCustomGeometry();
+        IModelState customModelState = blockModel.customData.getCustomModelState();
+        if (customModelState != null)
+        {
+            sprite = new ModelStateComposition(customModelState.apply(Optional.empty()).orElse(TRSRTransformation.identity()), sprite.getState(), sprite.isUvLock());
+        }
+        if (customModel != null)
+        {
+            model = customModel.bake(blockModel.customData, modelBakery, spriteGetter, sprite, format, blockModel.getOverrides(modelBakery, otherModel, spriteGetter, format));
+        }
+        else
+        {
+            if(!Attributes.moreSpecific(format, Attributes.DEFAULT_BAKED_FORMAT))
+            {
+                // TODO: Provide custom vanilla baking handler for special formats.
+                throw new IllegalArgumentException("can't bake vanilla models to the format that doesn't fit into the default one: " + format);
+            }
+            model = blockModel.bakeVanilla(modelBakery, otherModel, spriteGetter, sprite, format);
+        }
+
+        if (customModelState != null && !model.doesHandlePerspectives())
+        {
+            model = new PerspectiveMapWrapper(model, customModelState);
+        }
+
+        return model;
+    }
+
+    public static class ModelLoaderAdapter implements IModelLoader<ModelAdapter>
+    {
+        private final ICustomModelLoader parent;
+
+        public ModelLoaderAdapter(ICustomModelLoader parent)
+        {
+            this.parent = parent;
+        }
+
+        @Override
+        public void onResourceManagerReload(IResourceManager resourceManager)
+        {
+
+        }
+
+        @Override
+        public ModelAdapter read(JsonDeserializationContext deserializationContext, JsonObject modelContents)
+        {
+            if (!modelContents.has("model"))
+                throw new RuntimeException("ModelLoaderAdapter requires a 'model' key that points to a model file.");
+
+            String modelLocation = modelContents.get("model").getAsString();
+
+            try
+            {
+                IUnbakedModel model = parent.loadModel(new ResourceLocation(modelLocation));
```

Comment:
This doesn't check `accepts` anywhere which could cause strange problems

Reply:
I assumed the person choosing the loader would know what loader to use, but on second thought...